src/components/PostDetails.tsx

```diff
@@ -1,117 +1,147 @@
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const { id, title, body } = post;
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [commentsLoadError, setCommentsLoadError] = useState('');
+  const [commentsDelError, setCommentsDelError] = useState<number[]>([]);
+  const [addNewComment, setAddNewComment] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    if (commentsLoadError) {
+      setCommentsLoadError('');
+    }
+
+    if (comments) {
+      setComments(null);
+    }
+
+    if (addNewComment) {
+      setAddNewComment(false);
+    }
+
+    client.get<Comment[]>(`/comments?postId=${id}`)
+      .then(setComments)
+      .catch(() => setCommentsLoadError("Can't load comments for current post"))
+      .finally(() => setLoading(false));
+  }, [post]);
+
+  const handleDeleteComment = (commentId: number) => {
+    client.delete(`/comments/${commentId}`)
+      .then(() => {
+        setComments(
+          current => current?.filter(item => item.id !== commentId) || null,
+        );
+        setCommentsDelError([]);
+      })
+      .catch(() => setCommentsDelError(current => [...current, commentId]));
+  };
+
+  const handleAddNewComment = (newComment: Comment) => {
+    setComments(current => (current ? [...current, newComment] : [newComment]));
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${id}: ${title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {body}
           </p>
         </div>
 
-        <div className="block">
+        {loading && (
           <Loader />
+        )}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+        {!loading && (
+          <div className="block">
+            {commentsLoadError && (
+              <div
+                className="notification is-danger"
+                data-cy="PostsLoadingError"
+              >
+                {commentsLoadError}
+              </div>
+            )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+            {comments && (!comments.length ? (
+              <p className="title is-4" data-cy="NoCommentsMessage">
+                No comments yet
+              </p>
+            ) : (
+              <>
+                <p className="title is-4">Comments:</p>
 
-          <p className="title is-4">Comments:</p>
+                {comments.map(comment => (
```

Comment:
```suggestion
                {comments.map(({name, email, id, body})=> (
```

you can easily destructure your obj like this